deploy: add CSI-Addons endpoint

nixpanic opened the pull request:
The --csi-addons-endpoint= option has been added recently, but was not
configured in the deployment files yet. The socket was incorrectly
created as `/csi-addons.sock`, by correcting the URL to the socket, the
socket now gets created as `/csi/csi-addons.sock` in the same directory
as other sockets.

If an endpoint is a UNIX Domain Socket, the format needs to be
`unix:///path/to/socket`. The `unix://` URL format allows an
authentication provider to be added directly after the `//`. If there is
no authentication provider needed, the field can remain empty. After the
authetication provider, the full path needs to be specified, starting
with a `/`. This means that URLs to a UDS should start with `unix:///`
in normal cases.

---

<details>
<summary>Show available bot commands</summary>

These commands are normally not required, but in case of issues, leave any of
the following bot commands in an otherwise empty comment in this PR:

- `/retest ci/centos/<job-name>`: retest the `<job-name>` after unrelated
  failure (please report the failure too!)
- `/retest all`: run this in case the CentOS CI failed to start/report any test
  progress or results

</details>

nixpanic committed:
deploy: add CSI-Addons endpoint

Deployments place all sockets for communicating with CSI components in
the shared `/csi` directory. The CSI-Addons socket was introduced
recently, but not configured to be in the same location (by default
placed in `/tmp`).

Signed-off-by: Niels de Vos <ndevos@redhat.com>

Madhu-1 commented on deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml:
```diff
@@ -118,6 +118,7 @@ spec:
             - "--type=rbd"
             - "--controllerserver=true"
             - "--endpoint=$(CSI_ENDPOINT)"
+            - "--csi-addons-endpoint=$(CSI_ADDONS_ENDPOINT)"
```
Need same changes in helm chart also

yati1998 replied:
@nixpanic and @Madhu-1 do you wish to add the changes for the helm chart in the same PR or different? Otherwise looks good to me.

nixpanic replied:
yes, I know! Just being lazy and updated it only here for easy copy/pasting in my test environment.

nixpanic replied:
Done now.

Madhu-1 commented on cmd/cephcsi.go:
```diff
@@ -129,7 +129,7 @@ func init() {
 
 	// CSI-Addons configuration
-	flag.StringVar(&conf.CSIAddonsEndpoint, "csi-addons-endpoint", "unix://tmp/csi-addons.sock", "CSI-Addons endpoint")
+	flag.StringVar(&conf.CSIAddonsEndpoint, "csi-addons-endpoint", "unix:///csi/csi-addons.sock", "CSI-Addons endpoint")
```
not related to this PR but looks like this new parameter missing in doc https://github.com/ceph/ceph-csi/blob/devel/docs/deploy-rbd.md#configuration

nixpanic replied:
right! I'll add it there  too

nixpanic replied:
this has been addressed too.

nixpanic committed:
doc: add --csi-addons-endpoint option to rbd deployment

Signed-off-by: Niels de Vos <ndevos@redhat.com>

nixpanic committed:
deploy: fix default URL for --csi-addons-endpoint option

The --csi-addons-endpoint= option has been added recently, but was not
configured in the deployment files yet. The socket was incorrectly
created as `/csi-addons.sock`, by correcting the URL to the socket, the
socket now gets created as `/tmp/csi-addons.sock` in the same directory
as other sockets.

If an endpoint is a UNIX Domain Socket, the format needs to be
`unix:///path/to/socket`. The `unix://` URL format allows an
authentication provider to be added directly after the `//`. If there is
no authentication provider needed, the field can remain empty. After the
authetication provider, the full path needs to be specified, starting
with a `/`. This means that URLs to a UDS should start with `unix:///`
in normal cases.

Signed-off-by: Niels de Vos <ndevos@redhat.com>

yati1998 reviewed (commented):
LGTM, will wait for @Madhu-1 to review it.
